### PR TITLE
qutebrowser: don't use distutils

### LINKED ISSUE #326248: Build failure: qutebrowser
### Steps To Reproduce

1. build qutebrowser

### Build log

```
writing byte-compilation script '/build/tmp_j8r15cb.py'
/nix/store/z7xxy35k7620hs6fn6la5fg2lgklv72l-python3-3.12.4/bin/python3.12 /build/tmp_j8r15cb.py
Traceback (most recent call last):
  File "/build/tmp_j8r15cb.py", line 1, in <module>
    from distutils.util import byte_compile
ModuleNotFoundError: No module named 'distutils'
error: command '/nix/store/z7xxy35k7620hs6fn6la5fg2lgklv72l-python3-3.12.4/bin/python3.12' failed with exit code 1
make: *** [misc/Makefile:21: install] Error 1
```

### Additional context

Distutils since python 3.10 has been deprecated, possibly the issue here? Only ran into problems after nix flake --update on my end, so not sure if this is a global issue.

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.35, NixOS, 24.11 (Vicuna), 24.11.20240624.2893f56`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.3`
 - nixpkgs: `/nix/store/bi5zxc0v5g6ylygdwyqzh280sccg3ykb-source`
```

### PR DESCRIPTION
**Pull request by dotlambda**
## Description of changes
fixes https://github.com/NixOS/nixpkgs/issues/326248
closes #326464 cc @nrdxp 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/networking/browsers/qutebrowser/default.nix [hunk 1]
```diff
@@ -43,15 +43,17 @@ python3.pkgs.buildPythonApplication {
   buildInputs = [
     qtbase
     glib-networking
+  ] ++ lib.optionals stdenv.isLinux [
```
Comment (nrdxp) [inline pkgs/applications/networking/browsers/qutebrowser/default.nix:46]: did you actually test that it doesn't break the package? It was too long ago to rely soley on my memory so I'll have to try it.

File: pkgs/applications/networking/browsers/qutebrowser/default.nix [hunk 2]
```diff
@@ -83,7 +83,7 @@ python3.pkgs.buildPythonApplication {
     runHook preInstall
 
     make -f misc/Makefile \
-      PYTHON=${python3.pythonOnBuildForHost.interpreter} \
+      PYTHON=${(python3.pythonOnBuildForHost.withPackages (ps: with ps; [ setuptools ])).interpreter} \
```
Comment (nrdxp) [inline pkgs/applications/networking/browsers/qutebrowser/default.nix:88]: damn I tried that, but at the wrong layer I guess :sweat_smile: :facepalm: 

Good catch

**Review by nrdxp — COMMENTED** [review]
Overall it's good. I suppose one could argue for my version since using setup.py directly is discouraged, but also avoiding the patch here is desirable.

I'll have to build and test the wayland thing to ensure it doesn't break, but then I'll approve.

File: pkgs/applications/networking/browsers/qutebrowser/default.nix (hunk 1 above)
Comment (dotlambda) [reply pkgs/applications/networking/browsers/qutebrowser/default.nix:46]: Yes. On GNOME on Wayland.

**Review by nrdxp — APPROVED** [review]
It works, thanks

**Comment by nrdxp** [comment]
@dotlambda do you have the merge bit? If so feel free to merge imo, unless any of the other maintainers object.

**Pull request merged**